README.md

```diff
@@ -1,7 +1,154 @@
+<!---
+-- Copyright Kani Contributors
+-- SPDX-License-Identifier: Apache-2.0 OR MIT
+---> 
+
 ## rust-lean-models
 
 Lean models of various Rust libraries to facilitate Lean-based verification of Rust programs.
 
+
+## Description
+The library defines equivalent Lean versions of functions from the Rust standard library.
+This library is intended to be used for verifying string-manipulating Rust programs via Lean.
+
+The Lean implementation is based on the description of the Rust functions, which are published at https://doc.rust-lang.org/std.
+The library includes:
+- Definitions of functions equivalent to those from the Rust standard library
+- Proofs that these definitions are correct (i.e., match our understanding of the Rust original)
+- Supporting lemmas and theorems for proof construction
+
+## Installation
+
+To use the library, add the following lines to your `lakefile.lean`:
+
+` require rust-lean-models from git "https://github.com/model-checking/rust-lean-models" `
+
+Then do `lake update` in the terminal.
+
+## Usage
+### Translating Rust program 
+
+For any Rust built-in functions which are used in user code , map it with 
+the corresponding function name in the library (see the Tables below).
+
+### Proof Tutorial
+We demonstrate some proof techniques in Proof_Tutorial.lean through two simple programs that compute
```

Comment:
Please make Proof_Tutorial.lean a link to the actual file so that people can click on it.